sources/test/jest/test-env.ts

```diff
@@ -94,7 +98,77 @@ function setupExec(): void {
     )
 }
 
+class TestCacheImpl implements GradleEnvCacheImpl {
+    private readonly cacheContent: Record<string, Record<string, Uint8Array>> = {}
+
+    private _isAvailable: boolean = true
+
+    isAvailable(): boolean {
+        return this._isAvailable
+    }
+
+    enable(): void {
+        this._isAvailable = true
+    }
+
+    disable(): void {
+        this._isAvailable = false
+    }
+
+    async saveCache(paths: string[], key: string): Promise<GradleEnvCacheEntry> {
+        if (this.cacheContent[key]) {
+            throw new CacheEntryAlreadyExistsError(`Cache entry with key ${key} already exists`)
+        }
+
+        const fileContents = paths.map(async (path) => {
+            return await readFile(path)
+        })
+
+        let size = 0
+        const value: Record<string, Buffer<ArrayBufferLike>> = {}
+        for (const [path, contentsPromise] of zip(paths, fileContents)) {
+            const contents = await contentsPromise
+            value[path] = contents
+            size += contents.byteLength
+        }
+
+        this.cacheContent[key] = value
+        return { key, size }
+    }
+
+    async restoreCache(paths: string[], primaryKey: string, restoreKeys?: string[], options?: RemoteCacheDownloadOptions): Promise<GradleEnvCacheEntry | undefined> {
+        for (const key in [primaryKey, ...(restoreKeys ?? [])]) {
```

Comment:
Admittedly, this is a pretty reasonable code implementation if we swapped out the direct file read/write for use in a non-gh-action world.